### PR TITLE
Added ADAPTMap class in pfsspy.map

### PR DESCRIPTION
**Pull request by STBadman**
Referencing #166  . This commit adds an ADAPTMap custom class to pfsspy.map. This checks the map header has the field ("model" : 'ADAPT'), and if so, corrects the coordinate system and  date-obs meta data. 

If a standard adapt fits file with a 3d datacube is passed to this class, a warning that the data is 3D is raised and the first realization is sliced out to create the map. A preprocessng step to produce a MapSequence of ADAPT maps will be written up as an example (see again #166)

### CODE REVIEW

File: pfsspy/map.py [hunk 1]
```diff
@@ -22,3 +22,18 @@ def is_datasource_for(cls, data, header, **kwargs):
         """Determines if header corresponds to an GONG map."""
         return (str(header.get('TELESCOP', '')).endswith('GONG') and
                 str(header.get('CTYPE1', '').startswith('CRLN')))
+
+
+class ADAPTMap(sunpy.map.GenericMap):
+    def __init__(self, data, header, **kwargs):
+        header['date-obs'] = header['maptime']
+        header['t_obs'] = header['maptime']
```
[the remark this answers is not in the document]
Comment (STBadman) [reply pfsspy/map.py:30]: I agree - I think I was using ['t_obs'] to parse the map time before the sunpy.map.Map change. WIll remove.

File: pfsspy/map.py [hunk 2]
```diff
@@ -22,3 +22,18 @@ def is_datasource_for(cls, data, header, **kwargs):
...
+        header['ctype1'] = 'CRLN-CEA'
```
[the remark this answers is not in the document]
Comment (STBadman) [reply pfsspy/map.py:31]: Actually this is wrong, ADAPT is not "CEA", it's uniform in latitude. These should be "CRLN", "CRLT".

One check would be :

`(adaptMap.meta['CUNIT2'] == 'deg') & (adaptMap.meta['naxis2']*adaptMap.meta['CDELT2'] == 180)`

which confirms the lat units are reported as degrees, and the size of the lat axis * lat spacing adds up to 180 degrees. This returns false if the [Thompson (2006)](https://www.aanda.org/articles/aa/pdf/2006/14/aa4262-05.pdf) correction is applied to CDELT2 for example, but true for adapt maps I've tried so far.

File: pfsspy/map.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (STBadman) [reply pfsspy/map.py]: Removed in db65d0c

File: pfsspy/map.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (STBadman) [reply pfsspy/map.py]: Check for uniform lat binning for adapt maps before assigning ctype1 and ctype2 addressed in Removed in db65d0c

File: pfsspy/map.py [hunk 3]
```diff
@@ -22,3 +22,20 @@ def is_datasource_for(cls, data, header, **kwargs):
...
                 str(header.get('CTYPE1', '').startswith('CRLN')))
+
+
+class ADAPTMap(sunpy.map.GenericMap):
+    def __init__(self, data, header, **kwargs):
+        header['date-obs'] = header['maptime']
+        if not ((header['cunit2'] == 'deg') & 
+                (header['naxis2']*header['cdelt2'] == 180)) :
+            raise AssertionError("Latitude metadata doesn't add to 180deg")
+        header['ctype1'] = 'CRLN'
```
Comment (dstansby) [inline pfsspy/map.py:33]: I think this should be `CRLN-CAR` and `CRLT-CAR` to indicate that it's a plate-caree projection (ie. equally spaced in latitude)?

Comment (STBadman) [reply pfsspy/map.py]: Agreed, fixed in 27a633d

**Comment by dstansby** [comment]
Hi @STBadman - I was just checking some stuff in the commit history and I think you haven't set your local git email address to be the same as your github.com account one, which I think is why you're not showing up in the contributors list here: https://github.com/dstansby/pfsspy/graphs/contributors.

**Pull request merged**